### ISSUE TITLE
Document both directions of platform channel calls

### ISSUE THREAD

**Issue by xster** [flutter/flutter#32941]
https://flutter.dev/docs/development/platform-integration/platform-channels currently has an example of calling native via Dart. We should document the reverse as well. 

For instance, how do I return an exception?

**Comment by mariaszek9003** [comment]
That's right. I'm searching for example with platform channel messaging in reverse direction (from native to flutter) but I can't find anything.

**Comment by matthew-carroll** [comment]
@xster I'm a little confused about the goal of this ticket. The first statement seems to want documentation for native invoking a method in Dart. But the second statement about returning an exception seems independent of the direction of communication. Also, I think the `Result` object is already demonstrated in the docs, right?

**Comment by xster** [comment]
I meant how does native invoke something in Dart and Dart answering with a result or error. 

Then how does Dart invoke something in native and native answering with a result or error.

**Comment by matthew-carroll** [comment]
Isn't the Dart to native communication already documented?
https://flutter.dev/docs/development/platform-integration/platform-channels#example-calling-platform-specific-ios-and-android-code-using-platform-channels

**Comment by xster** [comment]
Right, and it looks like the reverse isn't. Invoke a thing from native, have Dart handle the method call and read its args etc.

**Comment by jmagman** [comment]
Related to (or maybe a duplicate of) https://github.com/flutter/flutter/issues/32942

**Comment by xster** [comment]
related but not the same issue

**Comment by xster** [comment]
would be nice to have both directions described for pigeon on that page too @gaaclarke
